tests/feeds/test_manual_snapshot_source.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+from telliot_feeds.feeds.snapshot_feed import snapshot_feed
+from telliot_feeds.utils.log import get_logger
+
+logger = get_logger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_snapshot_zero_response(monkeypatch):
+    monkeypatch.setattr("builtins.input", lambda: "0")
+    (value, _) = await snapshot_feed.source.fetch_new_datapoint()
+    assert type(value) is bool
+    assert value is False
+
+
+@pytest.mark.asyncio
+async def test_snapshot_one_response(monkeypatch):
+    monkeypatch.setattr("builtins.input", lambda: "1")
+    (value, _) = await snapshot_feed.source.fetch_new_datapoint()
+    assert type(value) is bool
+    assert value is True
+
+
+@pytest.mark.asyncio
+async def test_snapshot_false_response(monkeypatch):
+    monkeypatch.setattr("builtins.input", lambda: "false")
+    (value, _) = await snapshot_feed.source.fetch_new_datapoint()
+    assert type(value) is bool
+    assert value is False
+
+
+@pytest.mark.asyncio
+async def test_snapshot_true_response(monkeypatch):
+    monkeypatch.setattr("builtins.input", lambda: "true")
+    (value, _) = await snapshot_feed.source.fetch_new_datapoint()
+    assert type(value) is bool
+    assert value is True
+
+
+@pytest.mark.asyncio
+async def test_snapshot_error_response(monkeypatch, capsys):
```

Comment:
I think you should check for the error msg using `caplog` fixture here, bc it looks like you set input == to 5, then equal to 0, then you get input using `fetch_new_datapoint()`, so it never runs into the input of 5.

I could be wrong, but looks like you need to set input to 5, do fetch new datapoint, expect the proper console output in `caplog`, set input to 0, then fetch new datapoint, assert value is bool/False.

Reply:
I'm actually having trouble trying to figure this out, how to interrupt the while loop from the test.  With the monkeypatch, it will just get stuck in the loop if the input is wrong.  User is asked to input a value and so long that the value is wrong it will keep asking for an input.

Reply:
I used `unittest.mock` to provide different user inputs during consecutive while loops. And skipped the check for the CLI output bc I couldn't figure that out